File: scala/src/com/lut/scala/Day1/Student.scala
package com.lut.scala.Day1

import java.util

import scala.beans.BeanProperty

/**
  * @author qsg
  * @Date 2019/11/25 16:55
  * @describe 类中不能写静态的东西
  * @version 1.0
  **/
class Student(@BeanProperty var name:String, age:Int) {
  /*

  主构造器 在类名后面紧紧跟着
  从构造器 在类内部声明的构造器
           内部构造器用this 必须依附于其它构造器

   */

  def this(){
    this("jake",20);
    this.show
  }

  def this(name:String){
    this(name,18)
    this
  }
  private var gender:Boolean=_;

  def show:Unit=println(name+"_"+age)

  override def toString: String = gender+"_"+name+"_"+test

  def test():String="你好世界"

//  var v2 = util.TreeSet[Student]
  var v3=(x:Int,y:Int)=>x+y;

}

File: Hadoop/hdfs/src/main/java/com/dfec/hdfs/file_readerandwriter/sequence_file/stu/Student.java
package com.dfec.hdfs.file_readerandwriter.sequence_file.stu;

import org.apache.hadoop.io.Writable;

import java.io.DataInput;
import java.io.DataOutput;
import java.io.IOException;

/**
 * @author qsg
 * @version 1.0 v
 * @Classname Student
 * @Date 2020/7/23 16:35
 * @Copyright DFEC
 **/
public class Student implements Writable {
    //hadoop 序列化对象
    private Long id;
    private String name;
    private int age;
    //必须提供无参构造函数
    public Student() {
    }
    @Override
    public void write(DataOutput dataOutput) throws IOException {
        dataOutput.writeLong(this.getId());
        dataOutput.writeUTF(this.getName());
        dataOutput.writeInt(this.age);
    }

    @Override
    public void readFields(DataInput dataInput) throws IOException {
        this.id = dataInput.readLong();
        this.name = dataInput.readUTF();
        this.age = dataInput.readInt();
    }

    public Student(Long id, String name, int age) {
        this.id = id;
        this.name = name;
        this.age = age;
    }

    @Override
    public String toString() {
        return "Student{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", age=" + age +
                '}';
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}
